File: vendas-services/src/main/java/br/com/vendas/services/product/group/ProductGroupService.java
package br.com.vendas.services.product.group;

import java.util.List;

import br.com.vendas.domain.product.ProductGroup;
import br.com.vendas.services.support.ServiceResponse;

public interface ProductGroupService {
	
	/**
	 * Retorna todas os grupos de determinada filial
	 * @param empresaID
	 * @param filialID
	 * @return
	 */
	ServiceResponse<List<ProductGroup>> findAllByBranch(Long organizationID, Long branchID, Integer offset);
	
	

	/**
	 * Retorna todos os grupos que iniciem com a descrição passada no filter.
	 * @param description - 
	 * @param organizationID
	 * @param offset
	 * @param limit
	 * @return
	 */
	ServiceResponse<List<ProductGroup>> findAllByDescription(String description, Long organizationID,Long branchID, Integer offset);


	/**
	 * Salva ou atualiza o grupo de produtos
	 * @param productGroup
	 * @return
	 */
	ServiceResponse<ProductGroup> saveOrUpdate(ProductGroup productGroup) ;



}

File: vendas-web/src/main/webapp/app/js/ng/services/ProductGroupService.js
'use strict';

vendasApp.factory('ProductGroupService',function(Restangular){

	var productGroupEdit; 
	return {

		/**
		 * Salva ou atualiza o grupo de produtos
		 */
		save : function(productGroup){
			var productGroup_ = Restangular.all("v1").all('productGroup').all('save').post(productGroup);
			return productGroup_;
		},


		/**
		 * Retorna todas os grupos de produtos de determinada filial.
		 * @returns
		 */
		getAllByBranch: function(organizationID,branchID, offset){

			var parameters = {
					'organizationID' : organizationID, 
					'branchID' : branchID,
					'offset': offset
			};


			return Restangular.all("v1").all("productGroup").all("getAllByBranch").getList(parameters).then(function(result){
				var p = {};
				p.value = result.value;
				p.rowCount = result.rowCount;			
				return p;
			});
		}, 
		
		/**
		 * Retorna todas os grupos de produtos de determinada filial.
		 * @returns
		 */
		getAllByDescription: function(description,organizationID,branchID, offset){

			var parameters = {
					'description': description,
					'organizationID' : organizationID, 
					'branchID' : branchID,
					'offset': offset
			};


			return Restangular.all("v1").all("productGroup").all("getAllByDescription").getList(parameters).then(function(result){
				var p = {};
				p.value = result.value;
				p.rowCount = result.rowCount;			
				return p;
			});
		},


		setProductGroupEdition : function(productGroup){
			productGroupEdit = productGroup;
		},

		getProductGroupEdition : function(){
			return productGroupEdit;
		} 

	};
});